pds_doi_service/core/outputs/web_client.py

```diff
@@ -87,22 +89,53 @@ def submit_content(self, payload, url, username, password,
 
         return response.text
 
-    def query_doi(self, url, query, username, password,
-                  content_type=CONTENT_TYPE_XML):
+    def submit_content(self, payload, content_type=CONTENT_TYPE_XML):
+        """
+        Submits the provided payload to a DOI service endpoint via the POST
+        action.
+
+        Inheritors of DOIWebClient should pull any required endpoint specific
+        parameters (URL, username, password, etc...) from the configuration
+        util bundled with the class.
+
+        Inheritors should also take the extra step of parsing and returning any
+        Doi objects from the response text (if request was successful).
+
+        Parameters
+        ----------
+        payload : str
+            Payload to submit to the DOI service. Should only correspond
+            to a single DOI record.
+        content_type : str
+            The content type to specify the format of the payload, as well as
+            the format of the response from the endpoint.
+
+        Returns
+        -------
+        doi : Doi
+            Doi object parsed from the response text.
+        response_text : str
+            Body of the response text.
+
+        """
+        raise NotImplementedError(
+            f'Subclasses of {__name__} must provide an implementation for '
```

Comment:
I think you want `{self.__class__.__name__}` here, otherwise the exception message will be `Subclasses of web_client must provide an implementation for submit_content()`

Reply:
Yes indeed, will fix